packages/iota/test/validation/transactionEssence.spec.ts

```diff
@@ -0,0 +1,147 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+import { ED25519_ADDRESS_TYPE } from "../../src/models/addresses/IEd25519Address";
+import { UTXO_INPUT_TYPE } from "../../src/models/inputs/IUTXOInput";
+import { BASIC_OUTPUT_TYPE } from "../../src/models/outputs/IBasicOutput";
+import { TREASURY_OUTPUT_TYPE } from "../../src/models/outputs/ITreasuryOutput";
+import { ADDRESS_UNLOCK_CONDITION_TYPE } from "../../src/models/unlockConditions/IAddressUnlockCondition";
+import { validateInputs } from "../../src/validation/inputs/inputs";
+import { validateOutputs } from "../../src/validation/outputs/outputs";
+import { cloneTransactionEssence } from "./testUtils";
+import { mockTransactionEssence, protocolInfoMock, mockMaxDistintNativeTokens } from "./testValidationMocks";
+
+describe("Parents validation", () => {
+    it("should fail with inputs count must be greater then equal to 1 and less then equal to 128", () => {
+        const txEssence = cloneTransactionEssence(mockTransactionEssence);
+        txEssence.inputs = [];
+        const result = validateInputs(txEssence.inputs);
+
+        expect(result.isValid).toEqual(false);
+        expect(result.errors).toBeDefined();
+        expect(result.errors?.length).toEqual(1);
+        expect(result.errors).toEqual(expect.arrayContaining(
+            ["Inputs count must be between 1 and 128."]
+        ));
+    });
+
+    it("should fail with same transaction id and transaction output index in list of inputs", () => {
```

Comment:
"Should fail with..." tends to mean something a bit different in English. Like...
fail with error
fail with warning
fail with pleasure
I think what we are trying to write it "when" if should fail.
i.e. "should fail when transaction id and transaction output index pairs are not distinct"